src/core/reaction_ensemble.cpp

```diff
@@ -1,1724 +1,1970 @@
+  // repeated for each particle that occurs in that list
+  for (int i = 0; i < property_list.size(); i++) {
+    double charge = property_list[i].charge;
+    int type = property_list[i].type;
+#ifdef ELECTROSTATICS
+    // set charge
+    set_particle_q(property_list[i].p_id, charge);
+#endif
+    // set type
+    set_particle_type(property_list[i].p_id, type);
+  }
 }
 
 /**
-* Calculates the expression in the acceptance probability in the reaction ensemble
+* Calculates the expression in the acceptance probability in the reaction
+* ensemble
 */
-double ReactionEnsemble::calculate_boltzmann_factor_reaction_ensemble(single_reaction* current_reaction, double E_pot_old, double E_pot_new, std::vector<int>& old_particle_numbers){
-	/**calculate the acceptance probability in the reaction ensemble */
-	const double volume = m_current_reaction_system.volume;
-	const double factorial_expr=calculate_factorial_expression(current_reaction, old_particle_numbers.data());
-
-	const double beta =1.0/m_current_reaction_system.temperature_reaction_ensemble;
-	const double standard_pressure_in_simulation_units=m_current_reaction_system.standard_pressure_in_simulation_units;
-	//calculate boltzmann factor
-	return std::pow(volume*beta*standard_pressure_in_simulation_units, current_reaction->nu_bar) * current_reaction->equilibrium_constant * factorial_expr * exp(-beta * (E_pot_new - E_pot_old));
+double ReactionEnsemble::calculate_boltzmann_factor(
+    SingleReaction &current_reaction, double E_pot_old, double E_pot_new,
+    std::vector<int> &old_particle_numbers, int dummy_old_state_index,
+    int dummy_new_state_index,
+    bool dummy_only_make_configuration_changing_move) {
+  /**calculate the acceptance probability in the reaction ensemble */
+  const double factorial_expr = calculate_factorial_expression(
+      current_reaction, old_particle_numbers.data(), *this);
+
+  const double beta = 1.0 / temperature;
+  // calculate boltzmann factor
+  return std::pow(volume * beta * standard_pressure_in_simulation_units,
+                  current_reaction.nu_bar) *
+         current_reaction.equilibrium_constant * factorial_expr *
+         exp(-beta * (E_pot_new - E_pot_old));
 }
 
 /**
 *generic one way reaction
 *A+B+...+G +... --> K+...X + Z +...
-*you need to use 2A --> B instead of A+A --> B since in the last case you assume distinctness of the particles, however both ways to describe the reaction are equivalent in the thermodynamic limit
-*further it is crucial for the function in which order you provide the reactant and product types since particles will be replaced correspondingly! If there are less reactants than products, new product particles are created randomly in the box. Matching particles simply change the types. If there are more reactants than products, old reactant particles are deleted.
+*you need to use 2A --> B instead of A+A --> B since in the last case you assume
```

Comment:
Why is a reaction described right before the save_old_particle_numbers function?

Reply:
moved this comment to the reaction function